sdk/cognitiveservices/azopenai/examples_client_test.go

```diff
@@ -121,3 +125,44 @@ func ExampleClient_GetCompletionsStream() {
 		}
 	}
 }
+
+func ExampleClient_CreateImage() {
+	azureOpenAIKey := os.Getenv("AOAI_API_KEY")
+
+	// Ex: "https://<your-azure-openai-host>.openai.azure.com"
+	azureOpenAIEndpoint := os.Getenv("AOAI_ENDPOINT")
+
+	if azureOpenAIKey == "" || azureOpenAIEndpoint == "" {
+		fmt.Printf("Skipping example, environment variables missing\n")
+		return
+	}
+
+	keyCredential, err := azopenai.NewKeyCredential(azureOpenAIKey)
+
+	if err != nil {
+		panic(err)
+	}
+
+	client, err := azopenai.NewClientWithKeyCredential(azureOpenAIEndpoint, keyCredential, "", nil)
```

Comment:
@JeffreyRichter - image generation doesn't take a model deployment. It's making me think we might want to consider moving it per-request instead, and just modifying the URL dynamically on each request, rather than requiring it up-front.

What do you think about that?

Reply:
I would think this needs to be a per request param, otherwise we would be requiring the developer customer to create, for example, a client for chat completions and another client for embeddings (which may be ok, but I'm not sure if it is the experience we want to provide)

Reply:
Filed here: https://github.com/Azure/azure-sdk-for-go/issues/21102

I'll get some discussion going. I believe there's some discussion about how this is represented, even in C# (@trrwilson?) where the parameter for the 'model' is 'modelOrDeployment'.